EIPS/eip-5805.md

```diff
@@ -0,0 +1,401 @@
+Beyond simple monetary transactions, decentralized autonomous organizations are arguably one of the most important use cases of blockchain and smart contract technologies. Today, many communities are organized around a governance contract that allows users to vote. Among these communities, some represent voting power using transferable tokens ([EIP-20](./eip-20.md), [EIP-721](./eip-721.md), other). In this context, the more tokens one owns, the more voting power one has. Governor contracts, such as Compound's `GovernorBravo`, read from these "voting token" contracts to get the voting power of the users.
+
+Unfortunately, simply using the `balanceOf(address)` function present in most token standards is not good enough:
+- The values are not checkpointed, so a user can vote, transfer its tokens to a new account, and vote again with the same tokens.
+- A user cannot delegate their voting power to someone else without transferring full ownership of the tokens.
+
+These constraints have led to the emergence of voting tokens with delegation that the following logic:
+- Users can delegate the voting power of their tokens to themselves or a third party. This creates a distinction between balance and voting weight.
+- The voting weights of accounts are checkpointed, allowing lookups for past values at different points in time.
+- The balances are not checkpointed.
+
+This EIP is proposing to standardize the interface and behavior of these voting tokens.
+
+Additionally, the existing (non-standardized) implementations are limited to `block.number` based checkpoints. This choice causes many issues in a multichain environment, where some chains (particularly L2s) have an inconsistent or unpredictable time between blocks. This EIP also addresses this issue by allowing the voting token to use any time tracking function it wants, and exposing it so that other contracts (such as a Governor) can stay consistent with the token checkpoints.
+
+## Specification
+
+Following pre-existing (but not-standardized) implementation, the EIP proposes the following mechanism.
+
+Each user account (address) can delegate to an account of its choice. This can be itself, someone else, or no one (represented by `address(0)`). Assets held by the user cannot express their voting power unless they are delegated.
+
+When a "delegator" delegates its tokens voting power to a "delegatee", its balance is added to the voting power of the delegatee. If the delegator changes its delegation, the voting power is subtracted from the old delegatee's voting power and added to the new delegate's voting power. The voting power of each account is tracked through time so that it is possible to query its value in the past. With tokens being delegated to at most one delegate at a given point in time, double voting is prevented.
+
+Whenever tokens are transferred from one account to another, the associated voting power should be deducted from the sender's delegate and added to the receiver's delegate.
```

Comment:
I wonder if you want to specify whether this is `SHOULD` or "MUST" for handling the transfer behavior

Reply:
That's totally reasonable. If it's purposefully open to interpretation, I'd suggest
1. change the "should" to "MAY"
2. articulate it's purposefully open to interpretation / implementation choice in the `Rationale` section

Reply:
I'll stay with SHOULD .. because that what you should do unless you have a good reason not to, and not what you may do and you are not sure.

I'll add something in the Rationale section

Reply:
Sounds good